### ISSUE TITLE
unarchive /opt/path must be an existing dir

### ISSUE THREAD

**Issue by Delivion-GmbH** [ansible/ansible-modules-core#3725]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

unarchive

##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = /home/bamboo/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

[defaults]
remote_tmp = /tmp
ssh_args = -v
host_key_checking = False
callback_plugins = plugins

[ssh_connection]
control_path=%(directory)s/a%%h-%%p-%%r
ssh_args = -o StrictHostKeyChecking=no
pipelining=True
##### OS / ENVIRONMENT

centos-release-7-2.1511
##### SUMMARY

Instead of unarchiving the file as expected it is complaining about a missing dir. When using ansible 1.9 it is working as expected. We already tried it without pipelining=true but that didn't helped
##### STEPS TO REPRODUCE

```
Create an ansible task:
- name: Unzip CPAS external Java libs into {{cpas_dir}}/java
  unarchive: >
    copy=no
    src="/tmp/java-libs-13.zip"
    dest="{{cpas_dir}}"

cpas_dir is /opt/cpas
while cpas is a symlink to /opt/cpa-bootstrap
```

This is probably related to #932 and maybe to #2936 but since 932 was already closed we wanted to report this issue

##### EXPECTED RESULTS

```
changed: [10.xx.xx.xx] => {"changed": true, "check_results": {"unarchived": false}, "dest": "/opt/cpas", "extract_results": {"cmd": "/bin/unzip -o \"/tmp/java-libs-13.zip\" -d \"/opt/cpas\"", "err": "", "out": "Archive:  /tmp/java-libs-13.zip\n   creating: /opt/cpas/java/ext/\n   creating: /opt/cpas/java/jdbc/\n  inflating: /opt/cpas/java/jdbc/ojdbc6.jar  \n  inflating: /opt/cpas/java/ext/COPYRIGHT-jai_imageio.txt  \n  inflating: /opt/cpas/java/ext/LICENSE-jai_imageio.txt  \n   creating: /opt/cpas/java/ext/lib/\n   creating: /opt/cpas/java/ext/lib/amd64/\n  inflating: /opt/cpas/java/ext/lib/amd64/libmlib_jai.so  \n  inflating: /opt/cpas/java/ext/lib/amd64/libclib_jiio.so  \n   creating: /opt/cpas/java/ext/lib/ext/\n  inflating: /opt/cpas/java/ext/lib/ext/clibwrapper_jiio.jar  \n  inflating: /opt/cpas/java/ext/lib/ext/mlibwrapper_jai.jar  \n  inflating: /opt/cpas/java/ext/lib/ext/jai_imageio.jar  \n  inflating: /opt/cpas/java/ext/lib/ext/mail.jar  \n  inflating: /opt/cpas/java/ext/lib/ext/jai_core.jar  \n  inflating: /opt/cpas/java/ext/lib/ext/jai_codec.jar  \n  inflating: /opt/cpas/java/ext/THIRDPARTYLICENSEREADME-jai.txt  \n  inflating: /opt/cpas/java/ext/ENTITLEMENT-jai_imageio.txt  \n  inflating: /opt/cpas/java/ext/UNINSTALL-jai_imageio  \n  inflating: /opt/cpas/java/ext/THIRDPARTYLICENSEREADME-jai_imageio.txt  \n  inflating: /opt/cpas/java/ext/COPYRIGHT-jai.txt  \n  inflating: /opt/cpas/java/ext/DISTRIBUTIONREADME-jai.txt  \n  inflating: /opt/cpas/java/ext/DISTRIBUTIONREADME-jai_imageio.txt  \n  inflating: /opt/cpas/java/ext/LICENSE-jai.txt  \n  inflating: /opt/cpas/java/ext/UNINSTALL-jai  \n", "rc": 0}, "gid": 0, "group": "root", "handler": "ZipArchive", "mode": "0777", "owner": "root", "size": 25, "src": "/tmp/java-libs-13.zip", "state": "link", "uid": 0}
```
##### ACTUAL RESULTS

```
fatal: [172.xx.xx.xx]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"copy": "no", "dest": "/opt/cpas", "src": "/tmp/java-libs-13.zip"}, "module_name": "unarchive"}, "msg": "dest '/opt/cpas' must be an existing dir"}

Note: Since we are testing in a test-environment the IP is different
```

**Comment by Delivion-GmbH** [comment]
Just an update: This issue appears sometimes and sometimes not. Really strange

**Comment by jctanner** [comment]
@DevOPS-Delivion does disabling pipelining change the behavior in any way?

**Comment by Delivion-GmbH** [comment]
I think i tested that before opening the ticket and it didn't changed the behavior but if you want i can test it again to be sure

**Comment by abadger** [comment]
i think that this PR will fix things for you: https://github.com/ansible/ansible/pull/15999  Give it a try and I'll see about merging it to devel and the stable-2.1 branch (for release with the next 2.1.x release).

**Comment by abadger** [comment]
Merged to devel and stable-2.1 branches.  Please let us know if that doesn't fix things for you (preferably by opening a new bug and referencing this one -- we seldom see comments to closed bugs).

**Comment by AMeng** [comment]
This is resolved for me using the `stable-2.1` branch. Thank you!
